src/routes/ConsortiumManager/settings/users/general/PermissionSets/PermissionSets.js

```diff
@@ -77,6 +111,79 @@ export const PermissionSets = (props) => {
     contentData.find(({ id }) => id === selectedItemId)
   ), [contentData, selectedItemId]);
 
+  const searchParams = activeMember ? `?${TENANT_ID_SEARCH_PARAMS}=${activeMember}` : '';
+
+  const onCreate = () => {
+    history.push(`${PERMISSION_SET_ROUTES.CREATE}${searchParams}`);
+  };
+
+  const onCompare = () => {
+    history.push(`${PERMISSION_SET_ROUTES.COMPARE}${searchParams}`);
+  };
+
+  const onEdit = () => {
+    history.push(`${PERMISSION_SET_ROUTES.EDIT}/${selectedItemId}${searchParams}`);
+  };
+
+  const onCancel = () => {
+    history.push({
+      pathname: PERMISSION_SET_ROUTES.PERMISSION_SETS,
+      search: `?${TENANT_ID_SEARCH_PARAMS}=${tenantId}`,
+    });
+  };
+
+  const handleSuccess = (permissionName) => {
+    onCancel();
+
+    const messageId = 'ui-consortia-settings.consortiumManager.members.permissionSets.save.permissionSet.success';
+
+    queryClient.invalidateQueries({ tenantId });
+
+    return showCallout({
+      message: intl.formatMessage({ id: messageId }, { permissionName }),
+      type: 'success',
+    });
+  };
+
+  const { createPermission, removePermission, updatePermission } = useTenantPermissionMutations(tenantId, {
+    onError: handlePermissionsMutationError,
+  });
+
+  const onRemove = () => {
+    removePermission(selectedItemId, {
+      onSuccess: () => handleSuccess('remove'),
+    });
+  };
+
+  const getNormalizedPermissionSet = (values) => {
+    const filtered = omit(values, ['childOf', 'grantedTo', 'dummy', 'deprecated']);
+    const permSet = {
+      ...filtered,
+      mutable: true,
+      subPermissions: (values.subPermissions || []).map(p => p.permissionName),
+    };
+
+    return permSet;
+  };
+
+  const onSave = (values) => {
+    return createPermission(getNormalizedPermissionSet(values), {
+      onSuccess: () => handleSuccess(values.displayName),
+    });
+  };
+
+  const onSaveEdit = (values) => {
+    return updatePermission(getNormalizedPermissionSet(values), {
+      onSuccess: () => handleSuccess('update'),
+    });
+  };
+
+  const defaultProps = {
```

Comment:
you don't need this object on `PermissionSets` component level, I believe all of them can be defined in `ConsortiumPermissionSetsForm `